File: golden_ticket_enterprise/lib/entities/ticket_history.dart
class TicketHistory {
  final String action;
  final String actionMessage;
  final DateTime actionDate;

  TicketHistory({ required this.action, required this.actionMessage, required this.actionDate});

  factory TicketHistory.fromJson(Map<String, dynamic> json) {
    return TicketHistory(
      action: json['action'] ?? "No action provided",
      actionMessage: json['actionMessage'] ?? "No message provided",
      actionDate: DateTime.parse(json['actionDate']),
    );
  }
}

File: GoldenTicket/GoldenTicket/Entities/TicketHistory.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace GoldenTicket.Entities
{
    [Table("tblTicketHistory")]
    public class TicketHistory
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TicketHistoryID { get; set; }
        
        [Required]
        public int TicketID { get; set; }

        [ForeignKey("TicketID")]
        public Tickets? Ticket { get; set; } = null;
        [Required]
        public int ActionID { get; set; }

        [ForeignKey("ActionID")]
        public Action? Action { get; set; } = null;
        [Required]
        public string? ActionMessage { get; set; } = "";
        [Required]
        public DateTime? ActionDate { get; set; } = DateTime.Now;
    }

    public class TicketHistoryDTO {
        public string? Action { get; set;} = "";
        public string? ActionMessage { get; set;} = "";
        public DateTime? ActionDate { get; set;}
        
        public TicketHistoryDTO (TicketHistory ticketHistory){
            this.Action = ticketHistory.Action!.ActionName;
            this.ActionMessage = ticketHistory.ActionMessage;
            this.ActionDate = ticketHistory.ActionDate;
        }
    }
}
